### PR TITLE
Separate public API

### PR DESCRIPTION
**Pull request by johnsonr**
Consists entirely of moving files around. No functional change. However, imports will need to be reworked in downstream SDM instances.

Separates public APIs, from which `index.ts` is generated, into the `api` tree. This is a prelude to splitting this out into an `sdm-api` project.

The `internal` tree should be of no interest to end users. Other directories are library functionality, some of which may move out in time.

Also moves into the `pack` tree the various things like PCF and Checkstyle that will move out in time.

### CODE REVIEW

File: src/api/project/CachingProjectLoader.ts [hunk 1]
```diff
@@ -14,16 +14,16 @@
  * limitations under the License.
  */
 
-import { ProjectLoader, ProjectLoadingParameters, WithLoadedProject } from "./ProjectLoader";
+import { ProjectLoader, ProjectLoadingParameters, WithLoadedProject } from "../../spi/repo/ProjectLoader";
```
Comment (jessitron) [inline src/api/project/CachingProjectLoader.ts]: why repo instead of project? I see corresponding things in api/project

Comment (johnsonr) [reply src/api/project/CachingProjectLoader.ts]: Agreed. I can change this

File: test/handlers/events/delivery/artifact/gitHubReleaseArtifactStoreTest.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ import * as fs from "fs";
 import * as assert from "power-assert";
 import {
     GitHubReleaseArtifactStore,
-} from "../../../../../src/common/artifact/github/GitHubReleaseArtifactStore";
+} from "../../../../../src/internal/artifact/github/GitHubReleaseArtifactStore";
```
Comment (jessitron) [inline test/handlers/events/delivery/artifact/gitHubReleaseArtifactStoreTest.ts]: is everything not in /api or /spi effectively internal now?

File: test/handlers/events/delivery/scan/fingerprint/maven/mavenFingerprintTest.ts [hunk 3]
```diff
@@ -18,8 +18,8 @@ import { GitHubRepoRef } from "@atomist/automation-client/operations/common/GitH
 import { GitCommandGitProject } from "@atomist/automation-client/project/git/GitCommandGitProject";
 
 import * as assert from "power-assert";
-import { MavenFingerprinter } from "../../../../../../../src/common/delivery/code/fingerprint/maven/MavenFingerprinter";
-import { PushImpactListenerInvocation } from "../../../../../../../src/common/listener/PushImpactListener";
+import { PushImpactListenerInvocation } from "../../../../../../../src/api/listener/PushImpactListener";
+import { MavenFingerprinter } from "../../../../../../../src/pack/maven/MavenFingerprinter";
```
Comment (jessitron) [inline test/handlers/events/delivery/scan/fingerprint/maven/mavenFingerprintTest.ts]: is the pack stuff intended to be used from outside? ... perhaps each pack deserves a barrel

**Comment by jessitron** [comment]
what is the difference in meaning between /api and /spi, after this change?

File: test/handlers/events/delivery/build/local/npm/npmLogInterpreterTest.ts [hunk 4]
```diff
@@ -15,7 +15,7 @@
  */
 
 import * as assert from "power-assert";
-import { NpmLogInterpreter } from "../../../../../../../src/common/delivery/build/local/npm/npmLogInterpreter";
```
Comment (jessitron) [inline test/handlers/events/delivery/build/local/npm/npmLogInterpreterTest.ts]: i love that common is gone

File: test/handlers/events/delivery/scan/fingerprint/maven/mavenFingerprintTest.ts (hunk 3 above)
Comment (johnsonr) [reply test/handlers/events/delivery/scan/fingerprint/maven/mavenFingerprintTest.ts]: It's intended to move out in time

**Comment by johnsonr** [comment]
`api` is public interface for users
`spi` is interfaces to be implemented by implementers of SDMs.

**Pull request merged**